// Copyright 2015 The gRPC Authors
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.
syntax = "proto3";

option java_multiple_files = true;
option java_package = "io.grpc.examples.helloworld";
option java_outer_classname = "HelloWorldProto";
option objc_class_prefix = "HLW";

package helloworld;

// The greeting service definition.
service Greeter {
  // Sends a greeting
  rpc SayHello (HelloRequest) returns (HelloReply) {}
  rpc EditCot (CotRequest) returns (CotResponse) {}
}

// The request message containing the user's name.
message HelloRequest {
  string name = 1;
}

// The response message containing the greetings
message HelloReply {
  string message = 1;
}

// The request message containing the quotation parameters
message CotRequest {
  int32 usuarioId = 1;
  bool checkDescripcionLarga = 2;
  string observaciones = 3;
  double tipoCambio = 4;

  message GridRenglonCot {
    int32 removido = 1;
    int32 idDetalle = 2;
    double cantidad = 3;
    string notr = 4;
    bool requiereAutorizacion = 5;
  }

  repeated GridRenglonCot extraData = 5;
}

// The response message containing the quotation's create/edit operation results
message CotResponse {
  string valorRetorno = 1;
}
